File: data/java_skeleton/double-buffer/src/main/java/com/iluwatar/doublebuffer/Scene.java
/*
 * This project is licensed under the MIT license. Module model-view-viewmodel is using ZK framework licensed under LGPL (see lgpl-3.0.txt).
 *
 * The MIT License
 * Copyright © 2014-2022 Ilkka Seppälä
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package com.iluwatar.doublebuffer;

import java.util.List;
import lombok.extern.slf4j.Slf4j;
import org.apache.commons.lang3.tuple.Pair;

/**
 * Scene class. Render the output frame.
 */
@Slf4j
public class Scene {

  private final Buffer[] frameBuffers;

  private int current;

  private int next;

  /**
   * Constructor of Scene.
   */
  public Scene() {
  
  }

  /**
   * Draw the next frame.
   *
   * @param coordinateList list of pixels of which the color should be black
   */
  public void draw(List<? extends Pair<Integer, Integer>> coordinateList) {
      return;
  }

  public Buffer getBuffer() {
      return null;
  }

  private void swap() {
      return;
  }

}

File: data/c_sharp_skeleton/double-buffer/src/main/java/com/iluwatar/doublebuffer/Double-Buffer/Scene.cs
using System.Collections.Generic;
using Microsoft.Extensions.Logging;
using System;

namespace Com.Iluwatar.DoubleBuffer
{
    /// <summary>
    /// Scene class. Render the output frame.
    /// </summary>
    public class Scene
    {
        private readonly Buffer[] frameBuffers;

        private int current;

        private int next;

        /// <summary>
        /// Constructor of Scene.
        /// </summary>
        public Scene()
        {

        }

        /// <summary>
        /// Draw the next frame.
        /// </summary>
        /// <param name="coordinateList">List of pixels of which the color should be black</param>
        public void Draw(List<Tuple<int, int>> coordinateList)
        {
            return;
        }

        public Buffer GetBuffer()
        {
            return null;
        }

        private void Swap()
        {
            return;
        }
    }
}